lib/qesapdeployment.pm

```diff
@@ -480,4 +485,50 @@ sub qesap_ansible_script_output {
     return $output;
 }
 
+=head3 qesap_create_aws_credentials
+
+    Creates a AWS credentials file as required by QE-SAP Terraform deployment code.
+=cut
+
+sub qesap_create_aws_credentials {
+    my ($key, $secret) = @_;
+    my %paths = qesap_get_file_paths();
+    my $credfile = script_output q|awk -F\" '/aws_credentials/ {print $2}' | . $paths{qesap_conf_trgt};
+    save_tmp_file('credentials', join(' ', "[default]\naws_access_key_id", '=', $key,
+            "\naws_secret_access_key", '=', $secret, "\n"));
+    assert_script_run 'mkdir -p ~/.aws';
+    assert_script_run 'curl ' . autoinst_url . "/files/credentials -o $credfile";
+    assert_script_run "cp $credfile ~/.aws/credentials";
+}
+
+=head3 qesap_create_aws_config
+
+    Creates a AWS config file in ~/.aws as required by the QE-SAP Terraform & Ansible deployment code.
+=cut
+
+sub qesap_create_aws_config {
+    my %paths = qesap_get_file_paths();
+    my $region = script_output q|awk -F\" '/aws_region/ {print $2}' | . $paths{qesap_conf_trgt};
```

Comment:
Seems that all tests are able to get the region, why not to change qesap_create_aws_config to directly accept the region as argument?

Reply:
Same issue as with the credentials file: tests can have one region on a setting and another in the YAML configuration file. With this I'm just making sure that the `~/.aws/config` file generated by the method has the one that's actually used by terraform and ansible. If we decide to always replace the region in the YAML configuration file with whatever's in the **PUBLIC_CLOUD_REGION** setting, then we can change the above line to:
```
my $region = get_required_var('PUBLIC_CLOUD_REGION');
```

Reply:
Taking this into consideration, I'm adding an additional check in the function.

Reply:
we does not agree to always replace REGION in the yaml with exactly  PUBLIC_CLOUD_REGION, but we always replace it with something. We always have REGION controlled by the test code. But maybe I get your point: "As the only source of truth for the qe-sap-deployment is the config.yaml, and as the qe-sap-deployment expect some aws files to be in a specific disk place openQA test code should also use conf.yaml as source of truth to feel the gap and create the files." If it is so, maybe we can think to move it in the qesap glue script, the write, or at least the check, of the needed files.